Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -0,0 +1,570 @@
+        m_params.y_max = m_params.y_min + (m_params.ny-1)*spacing[1];
+        m_params.x_min = offset[2] + position[2]*spacing[2];
+        m_params.x_max = m_params.x_min + (m_params.nx-1)*spacing[2];
+    }
+#else
+    amrex::ignore_unused(lasy_file_name);
+#endif
+}
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::parse_binary_file(std::string binary_file_name)
+{
+    if(ParallelDescriptor::IOProcessor()){
+        std::ifstream inp(binary_file_name, std::ios::binary);
+        if(!inp) Abort("Failed to open binary file");
+        inp.exceptions(std::ios_base::failbit | std::ios_base::badbit);
+
+        //Uniform grid flag
+        char flag;
+        inp.read(&flag, 1);
+        if(!inp) Abort("Failed to read grid type from binary file");
+
+        //Grid points along t, x and y
+        inp.read(reinterpret_cast<char*>(&m_params.nt), sizeof(uint32_t));
+        inp.read(reinterpret_cast<char*>(&m_params.nx), sizeof(uint32_t));
+        inp.read(reinterpret_cast<char*>(&m_params.ny), sizeof(uint32_t));
+        if(!inp) Abort("Failed to read sizes from binary file");
+        if(m_params.nt <= 1) Abort("nt in binary file must be >=2");
+        if(m_params.nx <= 1) Abort("nx in binary file must be >=2");
+#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
+        if(m_params.ny <= 1) Abort("ny in binary file must be >=2 in 3D");
+#elif defined(WARPX_DIM_XZ)
+        if(m_params.ny != 1) Abort("ny in binary file must be 1 in 2D");
+#endif
+
+        //Coordinates
+        Vector<double> dbuf_t, dbuf_x, dbuf_y;
+        dbuf_t.resize(2);
+        dbuf_x.resize(2);
+#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
+        dbuf_y.resize(2);
+#elif defined(WARPX_DIM_XZ)
+        dbuf_y.resize(1);
+#endif
+        inp.read(reinterpret_cast<char*>(dbuf_t.dataPtr()),
+            dbuf_t.size()*sizeof(double));
+        inp.read(reinterpret_cast<char*>(dbuf_x.dataPtr()),
+            dbuf_x.size()*sizeof(double));
+        inp.read(reinterpret_cast<char*>(dbuf_y.dataPtr()),
+            dbuf_y.size()*sizeof(double));
+        if(!inp) Abort("Failed to read coords from binary file");
+
+        m_params.t_min = static_cast<amrex::Real>(dbuf_t[0]);
+        m_params.t_max = static_cast<amrex::Real>(dbuf_t[1]);
+        m_params.x_min = static_cast<amrex::Real>(dbuf_x[0]);
+        m_params.x_max = static_cast<amrex::Real>(dbuf_x[1]);
+#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
+        m_params.y_min = static_cast<amrex::Real>(dbuf_y[0]);
+        m_params.y_max = static_cast<amrex::Real>(dbuf_y[1]);
+#endif
+    }
+}
+
+std::pair<int,int>
+WarpXLaserProfiles::FromFileLaserProfile::find_left_right_time_indices(amrex::Real t) const
+{
+    int idx_t_right;
+    const auto t_min = m_params.t_min;
+    const auto t_max = m_params.t_max;
+    const auto temp_idx_t_right = static_cast<int>(std::ceil( (m_params.nt-1)*(t-t_min)/(t_max-t_min)));
+    idx_t_right = max(min(temp_idx_t_right, m_params.nt-1),1);
```

Comment:
I would suggest initializing `idx_t_right` only when you actually need it
```suggestion
    const auto t_min = m_params.t_min;
    const auto t_max = m_params.t_max;
    const auto temp_idx_t_right = static_cast<int>(std::ceil( (m_params.nt-1)*(t-t_min)/(t_max-t_min)));
    const int idx_t_right = max(min(temp_idx_t_right, m_params.nt-1),1);
```